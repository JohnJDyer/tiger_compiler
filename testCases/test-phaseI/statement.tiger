let
  var a: int;
  var b: array[10] of float;
in
  a := a;
  a := not(2);
  a := not(a);
  a := b[1];
  b[0] := not(1);
  b[1] := not(a);
  flush();
  while 1 do flush(); enddo;
  for a := 1 to 10 do flush(); enddo;
  for b := 1 to 10 do
    break;
  enddo;
  return 1;
end
