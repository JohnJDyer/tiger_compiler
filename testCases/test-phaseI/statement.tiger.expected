main:
	assign, a, a, 
	call, not, 2, $t0
	assign, a, $t0, 
	call, not, a, $t1
	assign, a, $t1, 
	array_load, $t2, b, 1
	assign, a, $t2, 
	array_load, $t3, b, 0
	call, not, 1, $t4
	assign, $t3, $t4, 
	array_load, $t5, b, 1
	call, not, a, $t6
	assign, $t5, $t6, 
	call, flush
loop_label0:
	breq, 1, 0, loop_label1
	call, flush
	goto, loop_label0, , 
loop_label1:
	assign, a, 1, 
loop_label2:
	brgt, a, 10, loop_label3
	call, flush
	add, a, 1, a
	goto, loop_label2, , 
loop_label3:
	assign, b, 1, 
loop_label4:
	brgt, b, 10, loop_label5
	goto, loop_label5, , 
	add, b, 1, b
	goto, loop_label4, , 
loop_label5:
	return, 1, , 
